vertexai/safety-settings-multimodal/safety-settings-multimodal.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+// [START aiplatform_gemini_safety_settings]
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"cloud.google.com/go/vertexai/genai"
+)
+
+func main() {
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	location := "us-central1"
+	modelName := "gemini-pro-vision"
+	temperature := 0.4
+
+	if projectID == "" {
+		log.Fatal("require environment variable GOOGLE_CLOUD_PROJECT")
+	}
+
+	cat, _ := partFromImagePath("https://storage.googleapis.com/cloud-samples-data/generative-ai/image/320px-Felis_catus-cat_on_snow.jpg")
+
+	// create a multipart (multimodal) prompt
+	prompt := []genai.Part{
+		genai.Text("say something nice about this "),
+		cat,
+	}
+
+	err := generateContent(os.Stdout, prompt, projectID, location, modelName, float32(temperature))
+	if err != nil {
+		fmt.Printf("unable to generate: %v\n", err)
+	}
+}
+
+// generateContent generates text from prompt and configurations provided.
+func generateContent(w io.Writer, prompt []genai.Part, projectID, location, modelName string, temperature float32) error {
+	ctx := context.Background()
+
+	client, err := genai.NewClient(ctx, projectID, location)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	model := client.GenerativeModel(modelName)
+	model.Temperature = temperature
+
+	// configure the safety settings thresholds
+	model.SafetySettings = []*genai.SafetySetting{
```

Comment:
Is there a difference in multi-modal setting of SafetySettings? i only see one sample listed for safety settings in the tracker.

Reply:
There's no difference in the SDK / API for SafteySettings. This (safety-settings-multimodal) is the proper sample for the current documentation set request.